### ISSUE TITLE
False positive for static return type

### ISSUE THREAD

**Issue by marcospassos** [thecodingmachine/phpstan-strict-rules#41]
The sniffs suggests replacing `static` type-hint by something like `self`:

```php
class Foo {
    /**
     * @return static
     */
	public function foo() : Foo {

	}
}
```

``` 
In method "Foo::foo", return type is type-hinted to "Foo" but the @return annotation says it is a "static(Foo)". Please fix the @return annotation. 
```
